### PR TITLE
[test] more specific feature_segwit test error messages and fixing incorrect comments

### PR DESCRIPTION
**Pull request by glozow**
Followup to [this](https://github.com/bitcoin/bitcoin/pull/15169/files#r303673472) comment on functional test feature_segwit.py verifying that unsigned witness transactions are invalid.

(1) Changes 8 error messages from "mandatory-script-verify-flag" to "non-mandatory-script-verify-flag" and with more specific error messages. 
(2) Edits comments that incorrectly describe the test, namely that the `v` variable corresponds to using P2WSH versus P2WPKH, not witness versions.

### CODE REVIEW

**Review by theStack — COMMENTED** [review]
Concept ACK -- it's definitely a good thing if error assertions get more specific.
Are there more assertion strings in the same file that could be improved in the course of this PR?

**Comment by gzhao408** [comment]
> Concept ACK -- it's definitely a good thing if error assertions get more specific.
> Are there more assertion strings in the same file that could be improved in the course of this PR?

Actually quite a few of the`feature_segwit.py` tests just have "mandatory-script-verify-flag". I'm a beginner so was just looking to address comments I see, but definitely seems a good idea to see if  the other tests in the file can be improved in the same way. 

Also noticed in `p2p_segwit.py`, most of the tests don't have error messages at all.

**Comment by amitiuttarwar** [comment]
Thanks for these test improvements! The code looks good to me, but I have some stylistic suggestions- 
* The PR description gets added to the merge commit (you can check out [recent commits](https://github.com/bitcoin/bitcoin/commits/master)).. so I'd recommend keeping it focused on describing the high level functionality of the PR. 
* Specific commit-level explanations can be on the commit messages themselves. Leaving the hashes out of the description also has the benefit of not needing to update every time you force push. I suspect your current commit messages are the product of some commits being squashed together, so I'd recommend revisiting to make them have a one-line description, followed by any additional description you think is helpful. here's an example of the rough format I like to use: https://github.com/bitcoin/bitcoin/pull/18038/commits/e82fdd329c812623c6e8a9022322fe43e1000c5b

**Comment by gzhao408** [comment]
> Thanks for these test improvements! The code looks good to me, but I have some stylistic suggestions-
> 
> * The PR description gets added to the merge commit (you can check out [recent commits](https://github.com/bitcoin/bitcoin/commits/master)).. so I'd recommend keeping it focused on describing the high level functionality of the PR.
> * Specific commit-level explanations can be on the commit messages themselves. Leaving the hashes out of the description also has the benefit of not needing to update every time you force push. I suspect your current commit messages are the product of some commits being squashed together, so I'd recommend revisiting to make them have a one-line description, followed by any additional description you think is helpful. here's an example of the rough format I like to use: [e82fdd3](https://github.com/bitcoin/bitcoin/commit/e82fdd329c812623c6e8a9022322fe43e1000c5b)

Thanks!! Cleaned up the description.

File: test/functional/feature_segwit.py [hunk 1]
```diff
@@ -117,8 +117,12 @@ def run_test(self):
 
         balance_presetup = self.nodes[0].getbalance()
         self.pubkey = []
-        p2sh_ids = []  # p2sh_ids[NODE][VER] is an array of txids that spend to a witness version VER pkscript to an address for NODE embedded in p2sh
-        wit_ids = []  # wit_ids[NODE][VER] is an array of txids that spend to a witness version VER pkscript to an address for NODE via bare witness
+        # p2sh_ids[NODE][VER=0] is an array of txids that spend to P2WPKH scripts to an address for NODE embedded in p2sh
```
Comment (amitiuttarwar) [inline test/functional/feature_segwit.py:120]: maybe instead of `VER` you can say `TYPE` since 0 -> P2WPKH and 1 -> P2WSH in both cases?

Comment (gzhao408) [reply test/functional/feature_segwit.py]: This is very true haha, changed!

**Comment by amitiuttarwar** [comment]
utACK b844fe742ccf683dd41d15e628423c130e9db1f1

File: test/functional/feature_segwit.py [hunk 2]
```diff
@@ -195,8 +195,8 @@ def run_test(self):
         self.log.info("Verify witness txs without witness data are invalid after the fork")
         self.fail_accept(self.nodes[2], 'non-mandatory-script-verify-flag (Witness program hash mismatch)', wit_ids[NODE_2][WIT_V0][2], sign=False)
         self.fail_accept(self.nodes[2], 'non-mandatory-script-verify-flag (Witness program was passed an empty witness)', wit_ids[NODE_2][WIT_V1][2], sign=False)
-        self.fail_accept(self.nodes[2], 'non-mandatory-script-verify-flag (Witness program hash mismatch)', p2sh_ids[NODE_2][WIT_V0][2], sign=False, redeem_script=witness_script(False, self.pubkey[2]))
-        self.fail_accept(self.nodes[2], 'non-mandatory-script-verify-flag (Witness program was passed an empty witness)', p2sh_ids[NODE_2][WIT_V1][2], sign=False, redeem_script=witness_script(True, self.pubkey[2]))
```
Comment (MarcoFalke) [inline test/functional/feature_segwit.py:199]: Why are those lines changed to remove the named arguments? That makes the tests harder to read, I think.

File: test/functional/feature_segwit.py [hunk 3]
```diff
@@ -172,13 +172,13 @@ def run_test(self):
...
+        self.fail_accept(self.nodes[0], "mandatory-script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_0][WIT_V0][0], sign=False)
+        self.fail_accept(self.nodes[0], "mandatory-script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_0][WIT_V1][0], sign=False)
         # unsigned with redeem script
-        self.fail_accept(self.nodes[0], "mandatory-script-verify-flag", p2sh_ids[NODE_0][WIT_V0][0], False, witness_script(False, self.pubkey[0]))
-        self.fail_accept(self.nodes[0], "mandatory-script-verify-flag", p2sh_ids[NODE_0][WIT_V1][0], False, witness_script(True, self.pubkey[0]))
+        self.fail_accept(self.nodes[0], "non-mandatory-script-verify-flag (Witness program hash mismatch)", p2sh_ids[NODE_0][WIT_V0][0], False, witness_script(False, self.pubkey[0]))
+        self.fail_accept(self.nodes[0], "non-mandatory-script-verify-flag (Witness program was passed an empty witness)", p2sh_ids[NODE_0][WIT_V1][0], False, witness_script(True, self.pubkey[0]))
```
Comment (MarcoFalke) [inline test/functional/feature_segwit.py:181]: nit: could add named arguments to the bool `False` to clarify its meaning?

File: test/functional/feature_segwit.py [hunk 4]
```diff
@@ -157,8 +157,8 @@ def run_test(self):
         self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][WIT_V1][0], True)  # block 427
 
         self.log.info("Verify unsigned p2sh witness txs without a redeem script are invalid")
-        self.fail_accept(self.nodes[2], "mandatory-script-verify-flag", p2sh_ids[NODE_2][WIT_V0][1], False)
-        self.fail_accept(self.nodes[2], "mandatory-script-verify-flag", p2sh_ids[NODE_2][WIT_V1][1], False)
+        self.fail_accept(self.nodes[2], "script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_2][WIT_V0][1], False)
+        self.fail_accept(self.nodes[2], "script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_2][WIT_V1][1], False)
```
Comment (MarcoFalke) [inline test/functional/feature_segwit.py:161]: Same here, since you touch this line anyway

**Review by maflcko — APPROVED** [review]
Concept ACK. Will test after the nits are addressed in some way.

File: test/functional/feature_segwit.py (hunk 2 above)
Comment (gzhao408) [reply test/functional/feature_segwit.py:199]: Oops! Must be leftover from me trying to shorten the lines or something. Fixed now.

**Comment by gzhao408** [comment]
Fixed named variables and added some for readability.

Also realized something that I should have fixed originally - changed the constant names from `WIT_V0` to `P2WPKH` and `WIT_V1` to `P2WSH`.  Doesn't change much, but adds some lines to the diff.

File: test/functional/feature_segwit.py (hunk 3 above)
Comment (gzhao408) [reply test/functional/feature_segwit.py]: added

File: test/functional/feature_segwit.py (hunk 4 above)
Comment (gzhao408) [reply test/functional/feature_segwit.py]: added

File: test/functional/feature_segwit.py [hunk 5]
```diff
@@ -151,14 +151,14 @@ def run_test(self):
...
+        self.skip_mine(self.nodes[2], wit_ids[NODE_2][P2WPKH][0], True)  # block 424
+        self.skip_mine(self.nodes[2], wit_ids[NODE_2][P2WSH][0], True)  # block 425
+        self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][P2WPKH][0], True)  # block 426
+        self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][P2WSH][0], True)  # block 427
 
         self.log.info("Verify unsigned p2sh witness txs without a redeem script are invalid")
-        self.fail_accept(self.nodes[2], "mandatory-script-verify-flag", p2sh_ids[NODE_2][WIT_V0][1], False)
-        self.fail_accept(self.nodes[2], "mandatory-script-verify-flag", p2sh_ids[NODE_2][WIT_V1][1], False)
+        self.fail_accept(self.nodes[2], "script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_2][P2WPKH][1], sign=False)
+        self.fail_accept(self.nodes[2], "script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_2][P2WSH][1], sign=False)
```
Comment (MarcoFalke) [inline test/functional/feature_segwit.py:161]: why are you removing the `mandatory-` prefix?

**Review by maflcko — APPROVED** [review]
Please add a newline in the commit message after the commit subject line and before the commit body. Otherwise the whole commit message will get parsed as a single (very long) line.

ACK 9b5710700eb4b201d1934744d9358a6f8cac39af 🎠

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 9b5710700eb4b201d1934744d9358a6f8cac39af 🎠
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhpqAv+IGmbmMqZTm8QyZm+vxcayWuYrMdiJLTfXFI9zCTNK76ypGKG+jNZruSX
7xIQglyc3knygWVmaS2IO+5ZjhbHdHegI9KnO9yuSeUGiQ8Zy7Plz3BxTJC4JGD2
BzkcTPonHbnJCiqn+E2Ayu17i+SV/WhAVEKVV2HdHQ77YLyZZ0FP1D5D2FB26O4a
J9mWJLRsziCnP6ZigaH5VEQsZuVEYnRbbf6boPesfgnZVkzkaO0knn0myh7ng0FH
iFFdO7IVKJSD4EVlL70LuRM6Z1nq0SP1guO3uqJI3j7ZenZvhAyx+FqG9VKJcMGc
o92hLwIBajbru66glqJAC3zdnn/uGl9NzjYsDm9weQcmTbaibgxmfN6fDU/TB8p2
AcV5i77h7tyWSwLuxzwsb38sEVmc+xEZ/YspXcJCuh+mGe0EVyUDkVgnusA9piSW
L57gsTEV7xI5PN/8Ziq896p78FwWNLPY/D0HdGZ8W9BdGO4/Otk/++fvSz5ClKn9
K4Ifnan+
=os7e
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `c4478981e5125a91a57c587c5dadab45769a9bcb45d4e1e34725040b83c50fb4  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108c4478981e5125a91a57c587c5dadab45769a9bcb45d4e1e34725040b83c50fb4f010d63cc24460bbcfe5172240baf4eb080d08fff0101c78570b97b204ffed3afa93d5cdef3708f1045e9f0464f008d025af9b7589c3650083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff01053b9aafd5b632d5dde763af786a64e2308f1045e9f0464f008dad238322cbd74d40083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010ea474536f0046ea46306894037dd8e9b08f1045e9f0464f008283d36d6d7afa8330083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010ed4342ced656197f6d27978de2cc48f208f1045e9f0464f0085404951100a43b420083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>

File: test/functional/feature_segwit.py (hunk 5 above)
Comment (gzhao408) [reply test/functional/feature_segwit.py]: Sorry about that, I'm such a rookie. Fixed now.

**Comment by MarcoFalke** [comment]
ACK 3c21db7b78 🍾

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 3c21db7b78 🍾
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUj6jgwAn+6koQ12kSueU0EwEin1DsU9lO3sqRd2ddipOJ6ukF6KZv7J8q80pGXC
IVFZFz2EiPi8YdQgu8flJspxVHmiPtLOjDYG13Hr11xcUSAW2826NGrDgpSfD682
+B4xDqAxuVi4XUCXBw+dfU/fqpOPrq8X7cQU8+yOsMc2dU6SfrNPAilzVDZgIw83
WsfZoaSAQBQFdMCH9FQDeevKb+HRsFk4qLOnyR1aNbc6SuHAdYml/E2qEtUT9Gxz
UBgUUPqDsbY7mp1Quw5SAtymzdlcdgf6huR8iziZqTohB3NOLU9fmO3Nroh2tJDF
X7ifqvRPFMFnu/qsfQHj/tYe9sKnpRZfw8rnfRrIASwdMas/TDnJ5YZ4w6TetygO
SZAnExKsVl9wYMtDyMZiIyM235QdStFvHkAo74Er/WZrx18MH8bOjF/bx+c8wRTo
Z47ZqdE2ZdMZapYCL1afCR4eaR3xykf777nNUJkrv9CNGRScHhmVhoumhgsNjwbm
XEO1uuyz
=uVmn
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `8a7bc4cb6ed0942427a32e552c127a6f2587b468b9577256a3b3e97a51ba8190  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401088a7bc4cb6ed0942427a32e552c127a6f2587b468b9577256a3b3e97a51ba8190f010b7fdf3ec65eb8469fdbe8e303577f8e208fff01029bf992985d17070558764326398e97d08f1045e9f2ff6f008678af8a2a837e2e20083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff01065d529c882b27c31964cab4108f157d008f1045e9f2ff6f00874728847ddf365380083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010a79ed1b5db934d296e89bcdbf6c394f408f1045e9f2ff7f00823823c65559ada880083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010da76d8dbcb1137985810eee2d607f18a08f1045e9f2ff7f0080d4bb32c9d0ef4d50083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>
